### DISCUSSION TITLE
Include markdown snippets

### DISCUSSION THREAD

**Discussion by dpaetzel** [srid/neuron#648] [Ideas]
I may be mistaken but I do not think that you can include other Zettels into a Zettel presently?

I would find this very handy, especially since I'm using the MathJax support extensively and start having the same preamble of the form
```
$$
\newcommand{…}{…}
(…)
\newcommand{…}{…}
$$
```
in more and more Zettels. It would be great if I could just write a Zettel with that preamble and then include that in every other Zettel that needs the definitions. For my use case, it would be enough to do this via the metadata as the definitions only need to be loaded at the top of the Zettel. Any thoughts on that? :slightly_smiling_face:

**Comment by dpaetzel** [comment]
Note that this is different to #328 as I might want to edit the included Zettel (e.g. add more definitions over time) which should then be reflected in the Zettels that import it.

**Comment by srid** [comment]
@TheMatten had some ideas on a "include" function, using the syntax from https://github.com/srid/neuron/issues/312#issuecomment-666313401 ... eg: 

```markdown
Hello world

{{ include | ./includes/preamble.md }}

Test
```

This would also allow us to embed comments widget (of #270) in every zettel. So I'm in favour of this feature, as long as the syntax for this is consistent with the proposal at #312.

---

Note that there was a similar feature request #184 - but it is not equivalent, as it wants to treat the file being included as a zettel (which isn't the case here).

**Comment by srid** [comment]
> which isn't the case here

Although this is not what you wrote. Could you clarify why you want the file to be included to be a zettel (instead of being a regular file under `./includes/`)?

**Comment by dpaetzel** [comment]
Thank you for your response! :slightly_smiling_face:  

> Could you clarify why you want the file to be included to be a zettel (instead of being a regular file under `./includes/`)?

Actually, it doesn't need to be a zettel but I do think it needs to get parsed in order for MathJax picking up on the `newcommand`s, doesn't it?

**Comment by srid** [comment]
Right. I imagine the include could happen _prior_ to parsing the Markdown text, so whatever you are including should automatically get parsed.

Renamed the issue accordingly.
